### PR TITLE
Add info about needed privileges for constraint commands

### PR DESCRIPTION
**Pull request by Hunterness**
Also update the info around IF NOT EXISTS.

4.3 constraint updates of https://github.com/neo4j/neo4j-documentation/pull/1157, the index section will be updated once it is restructured as well.

### CODE REVIEW

**Review by Lojjs — APPROVED** [review]
Haven't build it, but looks fine to me

**Comment by Hunterness** [comment]
For docs team: 
This adds links down to the administration chapter which might 
a) be affected by the link weirdness
b) need updating when that section of the administration chapter is moved
Not sure if either, but just as a heads-up :)

File: cypher/cypher-docs/src/docs/dev/ql/constraints/syntax.adoc [hunk 1]
```diff
@@ -17,10 +17,13 @@ If a name is not explicitly given, a unique name will be auto-generated.
 
 The create constraint command is optionally idempotent, with the default behavior to throw an error if you attempt to create the same constraint twice.
 With the `IF NOT EXISTS` flag, no error is thrown and nothing happens should a constraint with the same name or same schema and constraint type already exist.
-It may still throw an error should a conflicting index or constraint exist.
+It may still throw an error if conflicting data, indexes, or constraints exist.
+Examples of this would be nodes with missing properties, indexes with the same name, or constraints with same schema but a different constraint type.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/ql/constraints/syntax.adoc]:
```suggestion
Examples of this are nodes with missing properties, indexes with the same name, or constraints with same schema but a different constraint type.
```

File: cypher/cypher-docs/src/docs/dev/ql/constraints/syntax.adoc [hunk 2]
```diff
@@ -148,6 +153,7 @@ IS NODE KEY
 == Syntax for listing constraints
 
 List constraints in the database, either all or filtered on constraint type.
+This requires <<administration-security-administration-database-constraints, the `SHOW CONSTRAINT` privilege>>.
 
 The simple version of the command allows for a where clause and will give back the default set of output columns:
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/ql/constraints/syntax.adoc]:
```suggestion
The simple version of the command allows for a `WHERE` clause and will give back the default set of output columns:
```
While in this section, may as well fix this, if it's straightforward.

**Review by AlexicaWright — COMMENTED** [review]
Looks great! Just small comments.

**Pull request merged**